Libplanet/Store/DefaultStore.cs

```diff
@@ -629,6 +629,37 @@ HashDigest<SHA256> blockHash
             _statesCache.AddOrUpdate(blockHash, states);
         }
 
+        /// <inheritdoc/>
+        public override void PruneBlockStates<T>(
+            Guid chainId,
+            Block<T> until)
+        {
+            string[] keys = ListStateKeys(chainId).ToArray();
+            long untilIndex = until.Index;
+            foreach (var key in keys)
+            {
+                Tuple<HashDigest<SHA256>, long>[] stateRefs =
+                    IterateStateReferences(chainId, key, untilIndex, null, null)
+                        .OrderByDescending(tuple => tuple.Item2)
+                        .ToArray();
+                var dict = new Dictionary<HashDigest<SHA256>, List<string>>();
+                foreach ((HashDigest<SHA256> blockHash, long index) in stateRefs.Skip(1))
```

Comment:
It seems to be used once. how about using `IEnumerable`, not array?

```suggestion
                IEnumerable<Tuple<HashDigest<SHA256>, long>> stateRefs =
                    IterateStateReferences(chainId, key, untilIndex, null, null)
                        .OrderByDescending(tuple => tuple.Item2);
                var dict = new Dictionary<HashDigest<SHA256>, List<string>>();
                foreach ((HashDigest<SHA256> blockHash, long index) in stateRefs.Skip(1))
```